Sources/SnapshotPreviewsCore/SnapshotPreviewsCore.swift

```diff
@@ -186,7 +193,9 @@ public enum FindPreviews {
 
       return shouldInclude(name: name, excludedPreviewsSet: excludedPreviewsSet, previewsSet: previewsSet)
     }
-    return previewTypes.compactMap { preview -> PreviewType? in
+    let deviceName = ProcessInfo.processInfo.environment["SIMULATOR_DEVICE_NAME"] ?? ProcessInfo.processInfo.environment["SIMULATOR_MODEL_IDENTIFIER"]
+    return previewTypes
```

Comment:
Rather than doing the filtering here, let's do it when the previews get used, so in PreviewBaseTest `addMethods()` That way we won't also need to have the new initializer

Reply:
Correct me if I'm wrong, but in `PreviewBaseTest` we only use the `DiscoveredPreview` object, which doesn't have the device associated to it (particularly because `device` is individual preview specific, whereas `DiscoveredPreview` represents a grouping of Previews, similar to `PreviewType`.

I tried to see if I could pipe `device` in through all of the initializers, but it seems like it would result in a lot more changes than this PR (and not super straightforward)

plus I thought this location wasn't bad since we have the shouldInclude / excludedPreviewsSet right above it.

if it's easy to nab the device value directly within the `PreviewBaseTest` file though, happy to do it!